pcsx2/Gif.cpp

```diff
@@ -124,7 +235,13 @@ static u32 WRITERING_DMA(u32 *pMem, u32 qwc) {
 		else
 			qwc = std::min(qwc, 8u);
 	}
-	uint size = gifUnit.TransferGSPacketData(GIF_TRANS_DMA, (u8*)pMem, qwc*16) / 16;
+	uint size;
+	if (CHECK_GIFFIFOHACK) {
+		size = gif_fifo.write(pMem, qwc);
+	}
+	else {
+		size = gifUnit.TransferGSPacketData(GIF_TRANS_DMA, (u8*)pMem, qwc * 16) / 16;
+	}
```

Comment:
I think it'd be much cleaner/simpler to just initialize ``size``  using a ternary operator based on the status of ``CHECK_GIFFIFOHACK``